nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpResponse.java

```diff
@@ -194,7 +192,11 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         } catch (final ProcessException e) {
             session.transfer(flowFile, REL_FAILURE);
             getLogger().error("Failed to respond to HTTP request for {} due to {}", new Object[]{flowFile, e});
-            contextMap.complete(contextIdentifier);
+            try {
```

Comment:
Not a new issue but it could be improved around exception handling:
The typical (by convention) error handling flow is logging first, then sending the FF to FAILURE (in case of an exception in `session.transfer()`). Like in the last catch block at lines 210-211.

This catch and the others throughout onTrigger()  do it vice versa.
I would consider moving the
`session.transfer(flowFile, REL_FAILURE);`
lines just before the return statements.

Reply:
That makes sense, reordered this and other `session.transfer()` calls to occur after logging.